app/models/manageiq/providers/nuage/network_manager/event_catcher/qpid_container.rb

```diff
@@ -0,0 +1,15 @@
+# Workaround for bug https://issues.apache.org/jira/browse/PROTON-1791.
+# When the bug is resolved, we can use the Qpid::Proton::Container directly.
+class ManageIQ::Providers::Nuage::NetworkManager::EventCatcher::QpidContainer < Qpid::Proton::Container
+  def connection_driver(io, opts = nil, server = false)
+    # Store references to sockets so that we can later close them.
+    @sockets ||= []
+    @sockets << io
+
+    super(io, opts, server)
+  end
+
+  def close_sockets
+    (@sockets || []).each(&:close)
```

Comment:
Should you also clear the array after sockets are closed? Is the container reused later?

Reply:
Good point. Container is generally not reused (a new instance is always created [here](https://github.com/ManageIQ/manageiq-providers-nuage/blob/master/app/models/manageiq/providers/nuage/network_manager/event_catcher/stream.rb#L59)), but doing it properly is good idea anyway.